airbyte-ci/connectors/pipelines/pipelines/models/contexts/click_pipeline_context.py

```diff
@@ -0,0 +1,70 @@
+import sys
+from typing import Any, Callable, Optional
+
+import dagger
+from asyncclick import Context, get_current_context
+from dagger.api.gen import Client, Container
+from pydantic import BaseModel, Field, PrivateAttr
+
+from ..singleton import Singleton
+
+
+class ClickPipelineContext(BaseModel, Singleton):
+    dockerd_service: Optional[Container] = Field(default=None)
+    _dagger_client: Optional[Client] = PrivateAttr(default=None)
+    _click_context: Callable[[], Context] = PrivateAttr(default_factory=lambda: get_current_context)
+
+    @property
+    def params(self):
+        ctx = self._click_context()
+        click_obj = ctx.obj
+        click_params = ctx.params
+        command_name = ctx.command.name
+
+        # Error if click_obj and click_params have the same key
+        all_click_params_keys = [p.name for p in ctx.command.params]
+        intersection = set(click_obj.keys()) & set(all_click_params_keys)
+        if intersection:
+            raise ValueError(f"Your command '{command_name}' has defined options/arguments with the same key as its parent: {intersection}")
+
+        return {**click_obj, **click_params}
+
+    class Config:
+        arbitrary_types_allowed = True
```

Comment:
Why are we declaring a nested class here?

Reply:
This class was lifted from Aircmd with minor modifications. Didnt want to mess with an implementation that was working